src/persist/src/persist.proto

```diff
@@ -70,3 +69,20 @@ message ProtoU64Description {
     ProtoU64Antichain upper = 2;
     ProtoU64Antichain since = 3;
 }
+
+message ProtoBatchInline {
+    ProtoBatchFormat format = 1;
```

Comment:
I'm not totally happy with where this ProtoBatchFormat lives. as currently written, it's possible to read metadata out of a parquet file (so you have to know it's parquet) and then handle the ArrowKVTD variant. and vice versa.

one alternative is to push it into Proto{Unsealed,Trace}BatchMeta. this would mean we would know exactly what format the data of a blob is just from the reference stored in ProtoArrangement, which might be interesting. otoh, it makes the metadata bigger

another alternative is to give up on this enum being fully predictive of what the format of the blob is (my initial thought on what I want for the semantics) and instead just store something like KVTD, which simply tells the structure of the columns.

Reply:
The way this surfaces is that we hardcode our format in the `Codec` impl and then the decoding logic checks the format but will just throw up its hands when it notices that we didn't get what we expect, right? So right now it's only a sanity check?

Reply:
yeah. tho as a sanity check, we could just assert that the schema is exactly what we expect. the reason I added it is thinking forward to if we have to handle a migration where the types of the columns chang. imagine `[(K, V, T, D)]` (which is what this sortof is) to `[(K, [(V, [(T, D)])])]` which would only store the data for each unique key or val once. but it could also handle stuff like going from i32 offsets to i64. I can't come up with an example where it's not redundant with the actual schema, but I also like the explictness of it. dunno

Reply:
I think pushing the format into `ProtoUnsealedBatchMeta` makes the most sense to me unless you already tried it and the space overhead seemed prohibitive.

Reply:
Done